gtm/README.md

```diff
@@ -82,6 +84,44 @@ instantsearch.widgets.hits({
 > - Vue InstantSearch 2.1.0
 > - Angular InstantSearch 3.0.0
 
+Once these values are available in the hits template, you can retrieve them in GTM.
+
+1. Create a Custom JavaScript variable to find the element and name it `Find closest`
+
+```js
+function() {
+  return function(target, selector) {
+    while (!target.matches(selector) && !target.matches('body')) {
+      target = target.parentElement;
+    }
+
+    return target.matches(selector) ? target : undefined;
+  }
+}
+```
+
+This creates a variable called `{{Find closest}}` that contains the nearest element matching the selector.
+
+2. Create another Custom JavaScript variable and name it `ObjectIDClicked`
+
+```js
+function() {
+  var element = {{Find closest}}({{Click Element}}, '[data-insights-object-id]');
+
+  return typeof element !== 'undefined'
+    ? element.getAttribute('data-insights-object-id')
+    : undefined;
+}
+```
+
+This creates a variable called `{{ObjectIDClicked}}` that gives you the object ID of the hit clicked to reference it in the Insights method.
+
+You can repeat this process for other variables: `positions`, `queryId`, etc.
```

Comment:
I would avoid saying and rather explicite all variables that need to be declared with their related snippets

Reply:
I added more snippets in 72704023fec7d1f99fa3604afcb7b6ea91f520b5. Let me know if you think that's enough.